### PR TITLE
Force colors for local build tasks runnning under lerna

### PR DESCRIPTION
**Pull request by heliocliu**
Most tools auto-detect terminal color support, but lerna's run environment messes that up.  Force colors for build tasks that are typically run under lerna.
-For tests, pass along a `--color` flag from the root package.json
-For compile, force pretty format using the shared `ts-common-config.json`
-For lint, pass along a `--color` flag from the root package.json

### CODE REVIEW

**Comment by tylerbutler** [comment]
> because the files to lint must be passed as the last CLI arg to eslint.

I don't think that's true as of ESLint 7. From the [release notes](https://eslint.org/blog/2020/05/eslint-v7.0.0-released#improved-developer-experience):

> ESLint will now lint files with extensions other than `.js` if they are explicitly defined in `overrides[].files` - no need to use the `--ext` flag!

**Comment by heliocliu** [comment]
> 
> 
> > because the files to lint must be passed as the last CLI arg to eslint.
> 
> I don't think that's true as of ESLint 7. From the [release notes](https://eslint.org/blog/2020/05/eslint-v7.0.0-released#improved-developer-experience):
> 
> > ESLint will now lint files with extensions other than `.js` if they are explicitly defined in `overrides[].files` - no need to use the `--ext` flag!

Oh, I meant the `src` part of `eslint --format stylish src` for example.  I was getting errors when trying to do `eslint --format stylish src --color` because lerna sticks the args at the end, but it looks like I just screwed that up somehow and it actually works, so thanks for pointing this out and prompting me to check again 😅

File: package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "clean:r11s": "cd server/routerlicious && npm run clean",
     "postinstall": "npm run postinstall:lerna",
     "layer-check": "fluid-layer-check --info tools/build-tools/data/layerInfo.json",
-    "lint": "lerna run lint --no-sort --stream",
-    "lint:fix": "lerna run lint:fix --no-sort --stream",
+    "lint": "lerna run lint --no-sort --stream -- -- -- --color",
+    "lint:fix": "lerna run lint:fix --no-sort --stream -- -- -- --color",
```
Comment (tylerbutler) [inline package.json:32]: hahaha, why so many `--`? Isn't just one enough? Ohhh, this is because of all the npm script nesting, right?

**Review by tylerbutler — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (heliocliu) [reply package.json:32]: Yeah my earlier failed attempt used 3x `--` which was... insufficient

**Pull request merged**